src/content/en/fundamentals/native-hardware/recording-audio/index.md

```diff
@@ -0,0 +1,235 @@
+project_path: /web/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: Most browsers can get access to the user's microphone.
+
+{# wf_updated_on: 2016-08-22 #}
+{# wf_published_on: 2016-08-23 #}
+
+# Recording Audio from the User {: .page-title }
+
+{% include "web/_shared/contributors/paulkinlan.html" %}
+
+Many browsers now have the ability to access video and audio input from the user. However,
+depending on the browser it might be a full dynamic and inline experience, or it could 
+be delegated out to another app on the user's device.
+
+## Start simple and progressively
+
+The easiest thing that we can do is simply ask the user for a pre-recorded file. This can be 
+done by creating a simple file input element and adding an `accept` filter that indicates
+we can only accept audio files and ideally we will get them directly from the microphone.
+
+```
+<input type="file" accept="audio/*;capture=microphone">
+```
+
+This method will work on all platforms. On desktop it will prompt the user to upload a file from
+the file system (ignoring `capture=microphone`). In Safari on iOS it will open up the microphone 
+app, allowing you to record audio and then send it back to the web page; on Android it will 
+give the user of which app to use record the audio in before sending it back to the web page.
+
+Once the user has finished recording and they are back in the web site. You need to somehow get
+ahold of the file data. You can get quick access to this data by attaching an `onchange` event 
+to the input element and then reading the `files` property of the event object.
+
+<pre class="prettyprint">
+&lt;input type="file" accept="audio/*;capture=microphone" id="recorder">
+&lt;audio id="player">&lt;/audio>
+&lt;script>
+  var recorder = document.getElementById('recorder');
+  var player = document.getElementById('player')'
+
+  recorder.addEventListener('change', function(e) {
+    var file = e.files[0]; 
+    // Do something with the audio file.
+    player.src = file;
+  });
+&lt;/script>
+</pre>
+
+Once you have access to the file you can do anything you want with it. For example, you can:
+
+* Attach it directly to an `<audio>` element so that you can play it
+* Download it to the user's device
+* Upload it to a server by attaching to an XMLHTTPRequest
+* Pass it through the Web Audio API and apply filters on to it  
+
+Whilst this method of getting access to audio data is ubiquitous, it is the least appealing option. We really want to get access
```

Comment:
I'm trying to encourage wrapping at 80 char or less if possible. Makes diff'ing easier in the future.